code/game/objects/structures/crates_lockers/closets/cardboardbox.dm

```diff
@@ -62,3 +64,34 @@
 			for(var/mob/M in viewers(src))
 				M.show_message("<span class='notice'>\The [src] has been cut apart by [user] with \the [WC].</span>", 3, "You hear cutting.", 2)
 			qdel(src)
+			return
+		if(istype(W, /obj/item/weapon/storage/fancy/crayons))
+			var/list/contents = W.contents
+			var/crayoncount = 0
+			for(var/C in contents)
+				if(istype(C, /obj/item/toy/crayon))
+					crayoncount += 1
+				continue
+			if(crayoncount == 6)
+				var/decalselection = input("Please select a decal") as null|anything in list("Atmospherics", "Bartender", "Barber", "Blueshield",	"Brig Physician", "Captain",
+					"Cargo", "Chief Engineer",	"Chaplain",	"Chef", "Chemist", "Civilian", "Clown", "CMO", "Coroner", "Detective", "Engineering", "Genetics", "HOP",
+					"HOS", "Hydroponics", "Internal Affairs Agent", "Janitor",	"Magistrate", "Mechanic", "Medical", "Mime", "Mining", "NT Representative", "Paramedic", "Pod Pilot",
+					"Prisoner",	"Research Director", "Security", "Syndicate", "Therapist", "Virology", "Warden", "Xenobiology")
+				if(!decalselection)
+					return
+				if(user.incapacitated())
+					return
+				if(W == user.get_active_hand() && Adjacent(usr))
```

Comment:
I applied the host of suggested alterations to the msot recent commit with the exception of employing the W.storage_slots option because during compile it kept throwing an "undefined var" error, so I went with < 6 presuming that rayon box storage slots aren't going to change in the near or distant future.

I also applied a failure message for moving too far away in the interest of being thorough.